### PR TITLE
LRQA 69427

### PR DESCRIPTION
**Pull request by guivilarim**
Related Issue: https://issues.liferay.com/browse/LRQA-69427

- LRQA-69427 Add test stubs
- LRQA-69427 Create assertDataInRichTextField and inputDataInRichTextField macros
- LRQA-69427 Add CanNotPublishBlankRequiredField test
- LRQA-69427 Add CanInputDataOnDuplicatedField test
- LRQA-69427 Add CanRemoveDuplicatedField test
- LRQA-69427 Add test descriptions

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

File: modules/apps/data-engine/data-engine-test/src/testFunctional/macros/DERenderer.macro [hunk 1]
```diff
@@ -24,7 +24,6 @@ definition {
 			value1 = "${value}");
 	}
 
-<<<<<<< HEAD
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/macros/DERenderer.macro]: @guivilarim, this kind of code shouldn't be present in any commit. You should keep it out of you commits. :wink:

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DERichTextField.testcase [hunk 2]
```diff
@@ -135,7 +135,7 @@ definition {
 			fieldName = "Rich Text");
 	}
 
-	@description = "This is a stubbed description"
+	@description = "This is a test for LRQA-69427. This test verifies that data inputted on the Rich Text field persists."
```
Comment (natocesarrego) [inline modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DERichTextField.testcase:138]: @guivilarim, will you use the same LRQA for all new Poshi tests? Is it how you guys in original Data Engine Team were proceeding?

Can we talk more about this procedure after our daily today?

File: modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DERichTextField.testcase [hunk 3]
```diff
@@ -136,12 +136,35 @@ definition {
...
+			content = "Rich Text",
+			fieldLabel = "Rich Text");
+
+		Button.clickPublish();
+
+		WebContentNavigator.gotoEditCP(webContentTitle = "Web Content Title");
 
+		DERenderer.assertDataInRichTextField(
+			content = "Rich Text",
+			fieldLabel = "Rich Text");
```
[the remark this answers is not in the document]
Comment (guivilarim) [reply modules/apps/data-engine/data-engine-test/src/testFunctional/tests/fields/DERichTextField.testcase]: @natocesarrego which macros are from other teams? Most of the macros i've used were already been used in our previous Data Engine's Tesmap and avoid repeating code in our DataEngine's .macros files. Do you think it's better rewrite the macros in our files?

**Comment by natocesarrego** [comment]
Hi @guivilarim, after our talk from yesterday, will you revisit this pull to perform any change?

Thanks in advance for your update :wink:

**Comment by natocesarrego** [comment]
@guivilarim, let's talk about this pull today after our daily. :wink: Please remind me about it at the end of our meeting.

**Comment by natocesarrego** [comment]
Please resend

**Pull request closed without merging**